### PR TITLE
React router v6

### PR DESCRIPTION
**Pull request by bjmccotter7192**
## Description
Creating this Draft PR to get eyes on the new updates to React Router V6. This PR has been updated to use the new version, and have made updates to the: 
- Chapter
- Router
- Page
- Serve-Examples Pages
- Types

All of these changes were needed to get the routing in a working state. With the change to V6 the relative pathing makes it so we can use wildcard pathing to pass the chapters or pages.

## Things To Consider
1. We need to still figure out how to best test this type of routing
2. While using the wildcard path the Chapter and Page components compose the <Routes> and <Route> components but if you have multiple chapters and pages on the page the `<Link to='some-path'/>` seems to be acting not quite right.
3. There is a function in the Chapter component that filters all other children that are not `Page` Components, this function may need to be reworked/rewritten to fit our need better.
4. The ESLint rules have been commented out in this Draft PR due to check in restricting and still need to figure out the best option for that.
5. In the types.ts file under the `routing` directory I had to use `any` for the children of the ChapterProps which will need to be updated to fix the Typing issues and also part of the rework of the function mentioned in number 3
6. **NOTE** I chose to not use `<Outlet />` to get this working now but that could be a good resource to help figure out the path to issues and enable us to keep Pages inside of the Chapter component.

## Original issue(s)
[department-of-veterans-affairs/va-forms-system-core#240](https://app.zenhub.com/workspaces/forms-library---platform-spike-team-61b0ae1f2cd3c30014e8a5b0/issues/department-of-veterans-affairs/va-forms-system-core/240)

## Testing done
- NOT DONE AND THERE ARE BROKEN TESTS THAT NEED TO BE FIXED

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link to the original github issue has been provided
- [ ] No sensitive information (i.e. PII/credentials/internal URLs etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by micahchiang** [comment]
## Description

This commits I just pushed update a lot of different things, and at the risk of making an already big PR bigger, my suggestion is to create follow on tickets for remaining work. I'll leave a review and comment on some things, but also briefly mention things here as well. 

## Upgrading to React Router V6
With an upgrade to react router v6, how form applications are build will undergo a fundamental change. Users will build their form applications with a composition based approach by managing `Route` components and creating custom React components that plug into them. A high level example of this approach may look like this: 

```
<FormRouter basename={props.basename} formData={props.initialValues}>
    <Route index element={<PersonalInformationPage />} />
    <Route path="/page-two" element={<ContactInformationPage />} />
</FormRouter>
```

## Things changed
- The `Router` has been renamed to `FormRouter` to eliminate any confusion around imports from `react-router-dom`
- The `Page` component no longer renders a `Route` component. Instead, it only provides a Formik wrapper that handles form data and form state management
- Most of the examples have been removed in favor of keeping the multi-page form example. Additional refinement here will be necessary
- The `FormRouter` component no longer renders a formik `Form` component, instead letting the `Page` component handle that. The reason for this is to use a single `FormikContext` to manage all form state across multiple pages.
- The `FormRouter` component no longer provides it's own React Context, in favor of using the one Formik provides. 

## Follow On Tasks
- Refactor the `Chapter` component, as it no longer aligns with how form applications will be composed
- Rewrite tests for the `FormRouter`, `Chapter`, and `Page` components
- Decide on how to extract formik form data out of `FormRouter` and into a parent application

File: examples/multi-page/page-one.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    TextField,
+    EmailField,
+    Page, 
+    DebuggerView
+  } from '@department-of-veterans-affairs/va-forms-system-core';
+
+export default function PersonalInformationPage() {
+    return (
+        <>
+            <Page title="Personal Information" nextPage="page-two">
```
Comment (micahchiang) [inline examples/multi-page/page-one.jsx:13]: With the new approach, users will import the `Page` component into their own react components and build their form pages using va-forms-system-core provided form fields.

File: examples/simple-form/index.jsx [hunk 2]
```diff
@@ -55,10 +48,7 @@ const checkboxProps = {
   ],
 };
 
-// eslint-disable-next-line @typescript-eslint/no-unsafe-call
-void defineCustomElements();
-
-const App = () => (
+const SimpleApp = () => (
```
Comment (micahchiang) [inline examples/simple-form/index.jsx:51]: I think this accidentally got reverted, but it shouldn't matter too much. My plan is to submit a PR that updates the simple example to use the form I demoed at the end of sprint demo on 4/27.

File: src/routing/Chapter.tsx [hunk 3]
```diff
@@ -8,14 +8,20 @@ import { ChapterProps } from './types';
...
   return (
-    <Route path={props.path}>
-      <div className="vads-u-display--flex vads-u-align-items--center vads-u-flex-direction--column">
-        <h1>{props.title}</h1>
-      </div>
-      <Switch>
-        <>{props.children}</>
-      </Switch>
-    </Route>
+    <div>
```
Comment (micahchiang) [inline src/routing/Chapter.tsx:16]: Updated to router v6 conventions, but I mentioned in my overall comment we will most likely need to rewrite this.

File: src/routing/Page.tsx [hunk 4]
```diff
@@ -1,42 +1,36 @@
...
-            </Formik>
-          </div>
-        )}
-      </FormContext.Consumer>
-    </Route>
+    <div>
+      <h2>{props.title}</h2>
+      <Form>
+        {props.children}
+        <button
```
Comment (micahchiang) [inline src/routing/Page.tsx:22]: We have a ticket, #39, which will address navigation more closely. For now, this is just an initial first pass.

File: examples/index.jsx [hunk 5]
```diff
@@ -1,26 +1,29 @@
...
-      </li>
-    </ul>
-  </div>
-);
+import '@department-of-veterans-affairs/component-library/dist/main.css';
+import { defineCustomElements } from '@department-of-veterans-affairs/component-library';
+void defineCustomElements();
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const initialValues = {
```
Comment (micahchiang) [inline examples/index.jsx:11]: Used as `initialValues` for formik. Ideally, these would be collected from a json-schema and passed in by our users. Specifying initial values helps maintain form state when navigating to different pages.

File: test/routing/Chapter.test.tsx [hunk 6]
```diff
@@ -7,7 +7,7 @@ import Page from '../../src/routing/Page';
 import Chapter from '../../src/routing/Chapter';
 import { createMemoryHistory } from 'history';
 
-describe('Routing - Chapter', () => {
+describe.skip('Routing - Chapter', () => {
```
Comment (micahchiang) [inline test/routing/Chapter.test.tsx:10]: skipping these tests for now as they'll need to be rewritten with the chapter component rewrite

File: test/routing/Page.test.tsx [hunk 7]
```diff
@@ -7,7 +7,7 @@ import Page from '../../src/routing/Page';
 import Chapter from '../../src/routing/Chapter';
 import { createMemoryHistory } from 'history';
 
-describe('Routing - Page', () => {
+describe.skip('Routing - Page', () => {
```
Comment (micahchiang) [inline test/routing/Page.test.tsx:10]: skipping these tests for now, will be done in follow on ticket.

File: test/routing/Router.test.tsx [hunk 8]
```diff
@@ -3,13 +3,13 @@ import { render } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 
 import { Link } from 'react-router-dom';
-import Router from '../../src/routing/Router';
+import FormRouter from '../../src/routing/Router';
 import Page from '../../src/routing/Page';
 
-describe('Routing - Router', () => {
+describe.skip('Routing - Router', () => {
```
Comment (micahchiang) [inline test/routing/Router.test.tsx:9]: Skipping these tests for now, will be done in a follow on ticket.

**Review by mlauren — APPROVED** [review]

**Review by harshil1793 — APPROVED** [review]

**Review by pdavies88 — APPROVED** [review]

**Pull request merged**